socs/agents/ibootbar/agent.py

```diff
@@ -146,7 +151,7 @@ def update_cache(get_result, names, timestamp):
     except TypeError:
         oid_cache['ibootbar_connection'] = {'last_attempt': time.time(),
                                             'connected': False}
-        raise ConnectionError('No SNMP response. Check your connection.')
+        raise TypeError('Unable to interpret SNMP response.')
```

Comment:
The oid_cache is not returned if you raise an error.  So there's no point in setting it inside this "except" clause.

You need to decide whether you'll report errors by raising exceptions, or through the return value.